src/vehicle_autodrive.cpp

```diff
@@ -1,25 +1,1202 @@
 #include "vehicle.h" // IWYU pragma: associated
 
+#include <algorithm>
+#include <array>
+#include <cmath>
+#include <cstddef>
+#include <cstdint>
+#include <functional>
+#include <map>
+#include <memory>
+#include <queue>
+#include <string>
+#include <unordered_set>
+#include <utility>
+#include <vector>
+
+#include "avatar.h"
 #include "character.h"
 #include "coordinates.h"
+#include "cuboid_rectangle.h"
+#include "debug.h"
+#include "enums.h"
 #include "game.h"
 #include "map.h"
+#include "map_iterator.h"
+#include "map_memory.h"
+#include "mapdata.h"
+#include "messages.h"
+#include "optional.h"
 #include "point.h"
+#include "tileray.h"
+#include "translations.h"
+#include "type_id.h"
+#include "veh_type.h"
+#include "vpart_position.h"
+
+/*
+ * This file contains code that allows a vehicle to be driven by an in-game character (most
+ * likely the player character) to some remote destination, following a path of adjacent overmap
+ * tile (OMT) coordinates, while avoiding any obstacles along the way.
+ *
+ * The main entry point is do_autodrive(), which is intended to be called from the do_turn() of
+ * a long-running activity. This will cause the driver character to perform one or more driving
+ * actions (as long as they have enough moves), which can be either steering or changing the
+ * cruise control speed. The driver may also do nothing if the vehicle is going in the right
+ * direction.
```

Comment:
From our earlier discussion, I know that it's difficult to determine exactly what the driver has time to do. Thus, it matters in what order the driving is done (speed change first, steering first, etc). I'd say the ideal would be speed decreases, then steering, then speed increases.

Reply:
Speed changes are free (0 time units).

Reply:
```suggestion
 * a long-running activity. This will cause the driver character to perform one or more driving
 * actions (as long as they have enough moves), which can be steering, changing the
 * cruise control speed, or both. The driver may also do nothing if the vehicle is going in the
 * right direction.
```

Reply:
I did say "one or more", but I guess the "either" is confusing? How about:
"which could be steering and/or changing the cruise control speed"

Reply:
> I did say "one or more", but I guess the "either" is confusing? How about:

Yes.

> "which could be steering and/or changing the cruise control speed"

```suggestion
 * a long-running activity. This will cause the driver character to perform one or more driving
 * actions (as long as they have enough moves), which can be steering and/or changing the
 * cruise control speed. The driver may also do nothing if the vehicle is going in the right
 * direction.
```

Reply:
Done.